File: 30 Days of Code/C#/05 - Day 4 - Class vs Instance.cs
// ========================
//       Information
// ========================

// Direct Link: https://www.hackerrank.com/challenges/30-class-vs-instance/problem
// Difficulty: Easy
// Max Score: 30
// Language: C#

// ========================
//         Solution
// ========================

using System;
using System.Collections.Generic;
using System.IO;

class Person {
	public int age;

	public Person(int initialAge) {
		// Add some more code to run some checks on initialAge
		if (initialAge > 0) {
			age = initialAge;
		}
		else {
			Console.WriteLine("Age is not valid, setting age to 0.");
			age = 0;
		}
	}
	public void amIOld() {
		// Determining if this person's age is old
		if (age < 13) {
			Console.WriteLine("You are young.");
		}
		else if (age < 18) {
			Console.WriteLine("You are a teenager.");
		}
		else {
			Console.WriteLine("You are old.");
		}
	}

	public void yearPasses() {
		// Incrementing the age of the person
		age++;
	}
}

class Solution {
	public static void Main(String[] args) {
		int T = int.Parse(Console.In.ReadLine());
		for (int i = 0; i < T; i++) {
			int age = int.Parse(Console.In.ReadLine());
			Person p = new Person(age);
			p.amIOld();
			for (int j = 0; j < 3; j++) {
				p.yearPasses();
			}
			p.amIOld();
			Console.WriteLine();
		}
	}
}

File: 30 Days of Code/Python/05 - Day 4 - Class vs Instance.py
# ========================
#       Information
# ========================

# Direct Link: https://www.hackerrank.com/challenges/30-class-vs-instance/problem
# Difficulty: Easy
# Max Score: 30
# Language: Python

# ========================
#         Solution
# ========================

class Person:
    def __init__(self, initialAge):
        if initialAge > 0:
            self.age = initialAge
        else:
            self.age = 0
            print("Age is not valid, setting age to 0.")
    def amIOld(self):
        if self.age < 13:
            print("You are young.")
        elif(self.age >= 13 and self.age < 18):
            print("You are a teenager.")
        else:
            print("You are old.")

    def yearPasses(self):
        self.age += 1

t = int(input())

for i in range(0, t):
    age = int(input())
    p = Person(age)
    p.amIOld()

    for j in range(0, 3):
        p.yearPasses()
    p.amIOld()
    print("")
